client/src/utils/fight/setupFight.ts

```diff
@@ -386,60 +381,79 @@ const setupFight: (
   await sound.play('background', { loop: true });
 
   // Get fighters animations
-  const fighters: AnimationFighter[] = fightFighters.map((fighter) => {
-    // Necessary since .team was added later
-    // TODO: Remove on release
-    if (!fighter.team) {
-      fighter.team = (fighter.master || fighter.id) === brute1.id ? 'L' : 'R';
-    }
-
-    const animationFighter: AnimationFighter = {
-      ...fighter,
-      hpBar: (fighter.master
-        ? undefined
-        : fighter.id === brute1.id
-          ? brute1HpBar
-          : fighter.id === brute2?.id
-            ? brute2HpBar
-            : fighter.id === boss?.id
-              ? bossHpBar
+  fighters = await Promise.all(
+    fightFighters.map(async (fighter) => {
+      // Necessary since .team was added later
+      // TODO: Remove on release
+      if (!fighter.team) {
+        fighter.team = (fighter.master || fighter.id) === brute1.id ? 'L' : 'R';
+      }
+
+      const animationFighter: AnimationFighter = {
+        ...fighter,
+        hpBar: (fighter.master
+          ? undefined
+          : fighter.team === brute1.team
+            ? team1HpBar
+            : fighter.team === brute2?.team
+              ? team2HpBar
+              : fighter.id === boss?.id
+                ? bossHpBar
+                : undefined) ?? undefined,
+        hpBarPhantom: (fighter.master
+          ? undefined
+          : fighter.team === brute1.team
+            ? team1PhantomHpBar
+            : fighter.team === brute2?.team
+              ? team2PhantomHpBar
+              : fighter.id === boss?.id
+                ? bossPhantomHpBar
+                : undefined) ?? undefined,
+        bustImage: (fighter.master
+          ? undefined
+          : fighter.id === brute1.id
+            ? brute1BustImg
+            : fighter.id === brute2?.id
+              ? brute2BustImg
+              : fighter.type === 'brute'
+                ? await createBustImage(fighter, renderer)
```

Comment:
This whole .map has been asynced only for this.
It will impact performance on mobile when more than 2 brutes are rendered.
Instead, revert the .map to be non asynced, and only create the 2 first busts before the .map, and generate the others only when needed.

Reply:
This would require to add the   renderer: RendererContextInterface, as argument for every fight step that can change displayed brutes.

Do you want me to do this or do you prefer if instead of my current approach of adding setHUDFocus to every step, I just check if fightStep have "t,b, or f" in the main loop, for less "encombrement" ?